Source/Cells/Views/FORMTextField.m

```diff
@@ -107,14 +115,31 @@ - (void)setEnabled:(BOOL)enabled
 {
     [super setEnabled:enabled];
 
+    enabledProperty = enabled;
+
     if (enabled) {
-        self.backgroundColor = [UIColor FORMFieldBackground];
-        self.layer.borderColor = [UIColor FORMBlue].CGColor;
-        self.textColor = [UIColor FORMDarkBlue];
+        self.backgroundColor = enabledBackgroundColor;
+        self.layer.borderColor = enabledBorderColor.CGColor;
+        self.textColor = enabledTextColor;
+    } else {
+        self.backgroundColor = disabledBackgroundColor;
+        self.layer.borderColor = disabledBackgroundColor.CGColor;
+        self.textColor = disabledTextColor;
+    }
+}
+
+- (void)setValid:(BOOL)valid
+{
+    _valid = valid;
+
+    if (!self.isEnabled) return;
```

Comment:
I think we agree on not having random `return`-statements in methods.

```objc
if (self.isEnabled) {
    if (valid) {
        self.backgroundColor = validBackgroundColor;
        self.layer.borderColor = validBorderColor.CGColor;
     } else {
        self.backgroundColor = [UIColor FORMLightGray];
        self.layer.borderColor = [UIColor FORMFieldDisabledText].CGColor;
        self.textColor = [UIColor grayColor];
        self.backgroundColor = invalidBackgroundColor;
        self.layer.borderColor = invalidBorderColor.CGColor;
     }
}
```

I think that reads better.